src/fu-history.c

```diff
@@ -334,6 +336,48 @@ fu_history_migrate_database_v6(FuHistory *self, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_history_migrate_database_v7(FuHistory *self, GError **error)
+{
+	gint rc;
+	/* rename table */
+	rc = sqlite3_exec(self->db,
+			  "ALTER TABLE hsi_history RENAME TO hsi_history_tmp",
```

Comment:
do we really need to do this to alter one column type?

Reply:
Unfortunately, sqlite does not allow to switch data types using a ALTER command. The table should be rename and create a new table with the modified schema then copy all the records to the new table. Finally, the renamed old table will be dropped.